plugins/node/opentelemetry-instrumentation-pg/src/instrumentation.ts

```diff
@@ -369,3 +364,12 @@ function handleConnectResult(span: Span, connectResult: unknown) {
       })
   );
 }
+
+function isObjectWithTextString(it: unknown): it is ObjectWithText {
+  return (
+    typeof it === 'object' &&
+    typeof (it as null | { text?: unknown })?.text === 'string'
+  );
+}
+
+type ObjectWithText = { text: string; [k: string]: unknown };
```

Comment:
We can't, unfortunately, because `pg.QueryConfig` makes a bunch of other promises to TS that the code hasn't actually verified. E.g., it says that the `name` property will be a string or undefined, which isn't necessarily true. Making these minimal ad-hoc types might seem a bit redundant, but it makes it much less likely that we'll accidentally lie to TS about what we've checked (esp as these types evolve over time), and it doesn't create the need for extra boilerplate casts or anything, since TS will compare the types structurally.